homeassistant/components/unifiprotect/sensor.py

```diff
@@ -782,3 +828,72 @@ def _async_update_device_from_protect(self, device: ProtectDeviceType) -> None:
         self._set_event_attrs(event)
         if event.end:
             self._async_event_with_immediate_end()
+
+
+class ProtectPTZSensor(ProtectDeviceEntity, SensorEntity):
+    """A sensor for Unifi PTZ Cameras.
+
+    It exposes the current preset as the state value and pan/tilt/zoom as state attributes.
+    """
+
+    device: Camera
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        data: ProtectData,
+        device: ProtectDeviceType,
+        description: EntityDescription,
+    ) -> None:
+        """Override the constructor to include the device name."""
+        super().__init__(data, device, description)
+
+        self._hass = hass
+
```

Comment:
Don't pass in hass to an entity.  Its already provided by the base entity class

It will be set as `self.hass` once the entity has been added to Home Assistant before `async_added_to_hass` is called.